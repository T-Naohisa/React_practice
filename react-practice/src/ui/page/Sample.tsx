import React from "react";

export const FirstPage = () => {
  return (
    <div>
      <div>学習用</div>
      <div>お題目1:hooks useState</div>
    </div>
  );
};
