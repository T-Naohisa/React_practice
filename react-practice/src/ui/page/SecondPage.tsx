import React from "react";

export const SecondPage = () => {
  return (
    <>
      <div>dummy</div>
    </>
  );
};
