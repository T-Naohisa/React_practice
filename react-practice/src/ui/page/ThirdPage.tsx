import React from "react";

export const ThirdPage = () => {
  return (
    <>
      <div>dummy</div>
    </>
  );
};
