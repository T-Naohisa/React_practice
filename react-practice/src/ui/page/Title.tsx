import React from "react";

export const Title = () => {
  return (
    <div>
      <h2>useCallback</h2>
    </div>
  );
};
