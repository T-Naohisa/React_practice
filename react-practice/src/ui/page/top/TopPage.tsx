import React from "react";

import { TopPageTemplate } from "ui/template/TopTemplate";
export const TopPage = () => {
  return (
    <>
      <TopPageTemplate />
    </>
  );
};
