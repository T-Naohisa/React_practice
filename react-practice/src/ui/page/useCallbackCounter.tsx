import React from "react";

type props = { text: string; countState: number };

export const Count = React.memo((props: props) => {
  console.log("Count child component", props.text);
  return (
    <p>
      {props.text}:{props.countState}
    </p>
  );
});
