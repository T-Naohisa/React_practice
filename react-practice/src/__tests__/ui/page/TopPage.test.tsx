import React from "react";
import renderer from "react-test-renderer";
import { TopPage } from "ui/page/TopPage";
import { BrowserRouter as Router } from "react-router-dom";

it("render correctly", () => {
  const tree = renderer
    .create(
      <Router>
        <TopPage />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
