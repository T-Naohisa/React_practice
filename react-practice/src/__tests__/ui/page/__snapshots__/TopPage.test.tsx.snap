// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
Array [
  <div
    className="herader"
  >
    Header Compoennt
  </div>,
  <div
    className="firstmain"
  >
    <p>
      初期画面のtop
    </p>
    <p>
      <a
        href="/currentweather"
        onClick={[Function]}
      >
        currentweather
      </a>
    </p>
    <p>
      <a
        href="/forecasteachthreehours"
        onClick={[Function]}
      >
        forecasteachthreehours
      </a>
    </p>
  </div>,
  <div
    className="footer"
  >
    Footer Compoennt
  </div>,
]
`;
